great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -376,6 +374,20 @@ def run(
 
         return expectation_suite
 
+    def add_rule(self, rule: Rule) -> None:
+        """
+        Add Rule object to existing profiler object by reconciling profiler rules and updating _profiler_config.
+        """
+        rules_dict: dict = {rule.name: rule.to_json_dict()}
+        effective_rules: List[Rule] = self.reconcile_profiler_rules(
+            rules=rules_dict,
+        )
```

Comment:
Ah I feel a little iffy about this.

In `reconcile_profiler_rules`, we do a bunch of work, get to a point where we create dicts, and then convert them to Rules. Here, we're taking that work and then going back a step to convert it back to dicts.

I see @alexsherstinsky's concern about maintainability but I really would prefer not to backtrack like that it feels wasteful no? Performance isn't everything so I can see an argument being made against it - I do think we can find a happy medium though.

On a semi-related note, do we even need "reconciliation" here? It's this heavyweight, complex function that makes a lot of sense for runtime overrides but adding a rule is **so much** simpler right? Why do I need to do this complex reconciliation if all I'm doing is conditionally popping a value and appending? It feels slight overengineered.